app/cdap/services/helpers.js

```diff
@@ -723,6 +723,19 @@ function santizeStringForHTMLID(str) {
   return str.replace(/[ \/]/g, '-');
 }
 
+/**
+ * Wrapper for useEffect that ensures the latest version of the callback will be used, unlike useEffect
+ */
+function useOnUnmount(unmountHandler) {
```

Comment:
can we add this to a different file? this helpers file is already very bloated, maybe a custom hooks folder that is more top level? or `services/hooks/useOnUnmount.ts` ? In the spirit of starting to organize things I think it would make sense to keep this separate so its easier to find.

Reply:
To any future code archeologists: we discussed this and decided to split the file after 6.5 release